### ISSUE TITLE
[Feedback] Carbon CSS Custom Properties

### ISSUE THREAD

**Issue by joshblack** [carbon-design-system/carbon#4030]
Hi there! 👋 We're looking for feedback as the Carbon Design System team looks to add CSS Custom Property support for our component styles. For most users, these changes should be opaque as our stylesheets will change in the following way:

```diff
.some-component {
-  color: $token-01;
+  color: var(--token-01, $token-01);
}
```

Given that every CSS Custom Property will have the correct fallback, these will be functionally equivalent.

However, if you are targeting users on IE11 then you may be surprised if we shipped this change given that IE11 does not support custom properties. We would like to know if users would expect us to generate the appropriate fallback, or if it would be okay for us to ship this change and their build toolchain handles the appropriate conversions.

If you have a second, could you:

- 👍 this comment if you believe we can move forward with shipping CSS Custom Properties without providing the fallback
- 👎 this comment if you believe we should ship the fallbacks with our Sass source code for CSS Custom Properties

**Comment by ghost** [comment]
I'd welcome a shift away from sass as compilation takes quite long and therefore slows down dev process. Also, MS left behind IE11 quite a while ago and is now moving towards Edge + Chromium.

**Comment by daphoenix** [comment]
Great to see support for CSS Custom properties. This does not preclude using Sass (early-bound declarations) combined with CSS Custom properties (late-bound declarations) - http://net-informations.com/faq/oops/binding.htm.

**Comment by joshblack** [comment]
Will be shipped behind a feature flag in v10.7!
